sumpy/symbolic.py

```diff
@@ -294,92 +307,103 @@ class SpatialConstant(prim.Variable):
+
         return SympyToPymbolicMapperBase.map_Symbol(self, expr)
 
-    def map_Pow(self, expr):  # noqa: N802
+    @override
+    def map_Pow(self, expr: sym.Pow) -> Expression:
         if expr.exp == -1:
-            return 1/self.rec(expr.base)
+            return 1 / self.rec_arith(expr.base)
         else:
             return SympyToPymbolicMapperBase.map_Pow(self, expr)
 
-    def map_Mul(self, expr):  # noqa: N802
-        num_args = []
-        den_args = []
+    @override
+    def map_Mul(self, expr: sym.Mul) -> Expression:
+        num_args: list[ArithmeticExpression] = []
+        den_args: list[ArithmeticExpression] = []
         for child in expr.args:
             if (isinstance(child, Pow)
                     and isinstance(child.exp, Integer)
                     and child.exp < 0):
-                den_args.append(self.rec(child.base)**(-self.rec(child.exp)))
+                den_args.append(self.rec_arith(child.base)**(-self.rec_arith(child.exp)))
             else:
-                num_args.append(self.rec(child))
+                num_args.append(self.rec_arith(child))
 
         return math.prod(num_args) / math.prod(den_args)
 
 
 class PymbolicToSympyMapperWithSymbols(PymbolicToSympyMapper):
-    def map_variable(self, expr):
+    @override
+    def map_variable(self, expr: prim.Variable) -> sym.Basic:
         if expr.name == "I":
             return sym.I
         elif expr.name == "pi":
             return sym.pi
         else:
             return PymbolicToSympyMapper.map_variable(self, expr)
 
-    def map_subscript(self, expr):
+    @override
+    def map_subscript(self, expr: prim.Subscript) -> sym.Basic:
         if isinstance(expr.aggregate, prim.Variable) and isinstance(expr.index, int):
             return sym.Symbol(f"{expr.aggregate.name}{expr.index}")
         else:
             self.raise_conversion_error(expr)
+            raise
 
-    def map_call(self, expr):
-        if expr.function.name == "hankel_1":
-            args = [self.rec(param) for param in expr.parameters]
-            args.append(0)
-            return Hankel1(*args)
-        elif expr.function.name == "bessel_j":
-            args = [self.rec(param) for param in expr.parameters]
-            args.append(0)
-            return BesselJ(*args)
-        else:
-            return PymbolicToSympyMapper.map_call(self, expr)
+    @override
+    def map_call(self, expr: prim.Call) -> sym.Basic:
+        function = expr.function
+        if isinstance(function, prim.Variable):
+            if function.name == "hankel_1":
+                args = [self.rec(param) for param in expr.parameters]
+                args.append(sym.sympify(0))
+                return Hankel1(*args)
+            elif function.name == "bessel_j":
+                args = [self.rec(param) for param in expr.parameters]
+                args.append(sym.sympify(0))
+                return BesselJ(*args)
+
+        return PymbolicToSympyMapper.map_call(self, expr)
```

Comment:
Is this correct? pyright was complaining that `expr.function.name` does not have  `name` because `function: Expression`, but it wasn't clear how they're constructed here. I'm guessing the CI will fail if this is wrong 😁

Reply:
LGTM.